### PR TITLE
Deploy API doc & package automatically via CI

### PR DESCRIPTION
**Pull request by laysakura**
See `README.md`'s change for what will happen by this PR.

## Testing

These are published by [CI job](https://app.circleci.com/pipelines/github/Idein/actfw-core/79/workflows/066b1d54-8b30-4077-b21f-8d41aa6b7b41).

- https://idein.github.io/actfw-core/latest/
- https://pypi.org/project/actfw-core/

### CODE REVIEW

File: docs/Makefile [hunk 1]
```diff
@@ -1,20 +0,0 @@
...
-# Put it first so that "make" without argument is like "make help".
-help:
-	@$(SPHINXBUILD) -M help "$(SOURCEDIR)" "$(BUILDDIR)" $(SPHINXOPTS) $(O)
-
-.PHONY: help Makefile
-
-# Catch-all target: route all unknown targets to Sphinx using the new
-# "make mode" option.  $(O) is meant as a shortcut for $(SPHINXOPTS).
-%: Makefile
-	@$(SPHINXBUILD) -M $@ "$(SOURCEDIR)" "$(BUILDDIR)" $(SPHINXOPTS) $(O)
```
Comment (laysakura) [inline docs/Makefile:20]: removed unnecessary file

File: docs/make.bat [hunk 2]
```diff
@@ -1,35 +0,0 @@
...
-)
-
-%SPHINXBUILD% -M %1 %SOURCEDIR% %BUILDDIR% %SPHINXOPTS% %O%
-goto end
-
-:help
-%SPHINXBUILD% -M help %SOURCEDIR% %BUILDDIR% %SPHINXOPTS% %O%
-
-:end
-popd
```
Comment (laysakura) [inline docs/make.bat:35]: removed unnecessary file

File: docs/conf.py [hunk 3]
```diff
@@ -60,7 +60,7 @@ def setup(app):
 # Add any paths that contain custom static files (such as style sheets) here,
 # relative to this directory. They are copied after the builtin static files,
 # so a file named "default.css" will overwrite the builtin "default.css".
-html_static_path = ['_static']
+html_static_path = []
```
Comment (laysakura) [inline docs/conf.py:63]: suppresses `sphinx-build` warning (we currently don't have any plan to put `_static` by ourselves).

File: .circleci/config.yml [hunk 4]
```diff
@@ -3,13 +3,36 @@ version: 2.1
 commands:
   install-python-dev-tools:
     steps:
+      # See: https://circleci.com/docs/ja/2.0/language-python/#%E4%BE%9D%E5%AD%98%E9%96%A2%E4%BF%82%E3%81%AE%E3%82%AD%E3%83%A3%E3%83%83%E3%82%B7%E3%83%A5
```
Comment (notogawa) [inline .circleci/config.yml:6]:
```suggestion
      # See: https://circleci.com/docs/2.0/language-python/#cache-dependencies
```

File: .circleci/config.yml [hunk 5]
```diff
@@ -3,13 +3,36 @@ version: 2.1
...
+      - restore_cache:
+          key: &cache-key-pipenv cache-pipenv-{{ .Environment.CIRCLE_JOB }}-{{ .Environment.CIRCLECI_CACHE_VERSION }}-{{ checksum "Pipfile" }}
```
Comment (notogawa) [inline .circleci/config.yml:8]: `Pipfile.lock` should be added to the repository, and use the checksum of it.

File: .circleci/config.yml [hunk 6]
```diff
@@ -33,40 +56,121 @@ jobs:
...
       - run:
-          name: Make wheel
+          name: sphinx-build
           command: |
-            pipenv run python setup.py bdist_wheel
+            pipenv run sphinx-build docs build/sphinx/html
+      - persist_to_workspace:
+          root: build/sphinx
+          paths:
+            - html
```
Comment (notogawa) [inline .circleci/config.yml:72]: plz keep api docs as artifacts by `store_artifacts`

**Review by notogawa — CHANGES_REQUESTED** [review]

File: .circleci/config.yml (hunk 6 above)
Comment (laysakura) [reply .circleci/config.yml:72]: for what reason?
want to check docs out of artifacts before merging?

Comment (notogawa) [reply .circleci/config.yml:72]: To make docs reviewable for PR reviewers without rebuilding it on their own environment.

Comment (laysakura) [reply .circleci/config.yml:72]: got it.

File: .circleci/config.yml (hunk 5 above)
Comment (laysakura) [reply .circleci/config.yml]: https://github.com/Idein/actfw-core/pull/2/commits/76d8f8c4c887d75296a431455777cc1240c033f4

File: .circleci/config.yml (hunk 6 above)
Comment (laysakura) [reply .circleci/config.yml:103]: 3a2b628

File: .circleci/config.yml [hunk 7]
```diff
@@ -26,47 +67,190 @@ jobs:
...
     steps:
       - checkout
-      - install-python-dev-tools
+      - pipenv-sync
       - run:
           name: nose2
           command: pipenv run nose2 -v
-
-  build-wheel:
+  test-skip-lock:
```
Comment (laysakura) [inline .circleci/config.yml:77]: new job. tested here: https://app.circleci.com/pipelines/github/Idein/actfw-core/86/workflows/2e675314-8f06-4bab-90af-2d93c1c13d88/jobs/298

File: .circleci/config.yml [hunk 8]
```diff
@@ -49,34 +67,51 @@ jobs:
...
           command: |
             pipenv run sphinx-build docs build/sphinx/html
+            tar cvjf build/sphinx/html.tar.bz2 -C build/sphinx html
       - persist_to_workspace:
           root: build/sphinx
           paths:
             - html
+      - store_artifacts:  # for reviewers to check generated doc
+          path: build/sphinx/html.tar.bz2
+          destination: html.tar.bz2
```
Comment (notogawa) [inline .circleci/config.yml]: plz store `html` dir. If it is archived like this, we must download it to view.

**Review by notogawa — CHANGES_REQUESTED** [review]

File: .circleci/config.yml (hunk 8 above)
Comment (laysakura) [reply .circleci/config.yml]: I didn't know we can view HTML artifacts rendered in browser (right?).
OK, I will store only html dir.

Comment (laysakura) [reply .circleci/config.yml]: 82c3c07
🎉 https://310-321225649-gh.circle-artifacts.com/0/html/index.html

Comment (notogawa) [reply .circleci/config.yml]: working https://310-321225649-gh.circle-artifacts.com/0/html/index.html

**Review by notogawa — APPROVED** [review]
LGTM

**Pull request merged**